### PR TITLE
SYNCT-262: Move audit log view

### PR DESCRIPTION
**Pull request by areklalo**

### CODE REVIEW

File: api/src/main/java/org/openmrs/module/sync2/api/validator/impl/SyncConfigurationInstanceIdValidator.java [hunk 1]
```diff
@@ -10,11 +11,20 @@
...
     public void validate(SyncConfiguration syncConfiguration, Errors errors) {
         if (StringUtils.isBlank(syncConfiguration.getGeneral().getLocalInstanceId())) {
             errors.addErrorCode(ERROR_CODE);
         }
+        if (parentInstanceUriIsEmpty()) {
+            errors.addErrorCode(PARENT_URI_ERROR);
+        }
+    }
+
+    private boolean parentInstanceUriIsEmpty() {
```
Comment (kmadej) [inline api/src/main/java/org/openmrs/module/sync2/api/validator/impl/SyncConfigurationInstanceIdValidator.java]: I see this method is used few times. Please move it to SyncUtils class

File: omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+		} else if (message != null && !message.getSuccess()) {
+			InfoErrorMessageUtil.flashErrorMessage(session, ui.message(RETRY_FAILED));
+			resultUrl = buildNewMessageUri(backPageIndex, message);
+		}
+
+		return "redirect:" + resultUrl;
+	}
+
+	private String buildAuditListUri(Integer backPageIndex) {
+		if (backPageIndex == null) {
```
Comment (kmadej) [inline omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java]: not a best idea to change value of parameter passed to method. Please try to do it different way

File: omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+			backPageIndex = 1;
+		}
+		return AUDIT_LIST_PAGE + "?pageIndex=" + backPageIndex;
+	}
+
+	private String buildNewMessageUri(Integer backPageIndex, AuditMessage message) {
+		if (backPageIndex == null) {
```
Comment (kmadej) [inline omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java]: similar situation

File: omod/src/main/java/org/openmrs/module/sync2/web/controller/Sync2AuditListController.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+			@RequestParam(value = SyncMessageUtils.ALERT_MESSAGE_MODEL, required = false) String alertMessage,
+			@RequestParam(value = "backPageIndex", required = false) Integer backPageIndex) {
+
+		model.addAttribute(CONFIGURATION_VALIDATION_ERRORS, getConfigurationValidationErrors());
+		model.addAttribute(CREATOR_IDS, getCreatorIdsWithoutOwnId());
+		model.addAttribute(LOCAL_INSTANCE_ID, SyncUtils.getLocalInstanceId());
+		model.addAttribute(SyncMessageUtils.SUCCESS_MESSAGE, success);
+		model.addAttribute(SyncMessageUtils.ALERT_MESSAGE_MODEL, alertMessage);
+
+		if (backPageIndex == null) {
```
Comment (kmadej) [inline omod/src/main/java/org/openmrs/module/sync2/web/controller/Sync2AuditListController.java]: move it model.addAttribute line 

backPageIndex == null ? 1: backPageIndex

File: omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncConflictRestController.java [hunk 5]
```diff
@@ -67,4 +71,14 @@
 			return new ResponseEntity<>(result, HttpStatus.BAD_REQUEST);
 		}
 	}
+
+	private String buildResolutionUrl(AuditMessage conflictMessage, String auditBackPage, Integer backPageIndex) {
+		if (StringUtils.isBlank(auditBackPage)) {
```
Comment (kmadej) [inline omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncConflictRestController.java]: do not change method parameters inside

File: api/src/main/java/org/openmrs/module/sync2/api/validator/impl/SyncConfigurationInstanceIdValidator.java (hunk 1 above)
Comment (areklalo) [reply api/src/main/java/org/openmrs/module/sync2/api/validator/impl/SyncConfigurationInstanceIdValidator.java]: Good idea, thanks.

File: omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java (hunk 2 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java]: Thanks.

File: omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java (hunk 3 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/sync2/page/controller/RetryPageController.java]: As above.

File: omod/src/main/java/org/openmrs/module/sync2/web/controller/Sync2AuditListController.java (hunk 4 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/sync2/web/controller/Sync2AuditListController.java]: Corrected.

File: omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncConflictRestController.java (hunk 5 above)
Comment (areklalo) [reply omod/src/main/java/org/openmrs/module/sync2/web/controller/rest/SyncConflictRestController.java]: As above.

**Pull request merged**